src/components/app/details/appDetails/SecurityVulnerabilityCard.tsx

```diff
@@ -37,96 +35,64 @@ export const SecurityVulnerabilityCard = ({
+        () => getSecurityScan({ appId, envId }),
+        [appId, envId],
+        isSecurityScanV2Enabled,
+    )
+
+    const [executionDetailsLoading, executionDetailsResponse, executionDetailsError, reloadExecutionDetails] = useAsync(
+        () => getLastExecutionMinByAppAndEnv(appId, envId),
+        [appId, envId],
+        !isSecurityScanV2Enabled,
+    )
+
     const handleOnCardClick = () => {
-        if (!cardLoading && !lastExecutionDetail.isError && !showScanDetailsModal) {
+        if (!cardLoading && !scanResultError && !executionDetailsError && !showScanDetailsModal) {
             // hack: added timeout to prevent click outside
             setTimeout(() => setShowScanDetailsModal(true), 100)
         }
     }
 
-    async function getScanData() {
-        if ((!appId || !envId) && !installedAppId) {
-            return
-        }
-        setAPILoader(true)
-        try {
-            const { result } = await (isSecurityScanV2Enabled
-                ? getSecurityScan({ appId, envId, installedAppId })
-                : getLastExecutionMinByAppAndEnv(appId, envId))
-            let isScanFailed = false,
-                isScanInProgress = false
-            if (isSecurityScanV2Enabled) {
-                isScanFailed = result.codeScan?.status === 'Failed' || false
-                isScanInProgress = result.codeScan?.status === 'Progressing' || false
-                if (!isScanFailed && result.imageScan?.vulnerability?.list?.length) {
-                    for (const element of result.imageScan.vulnerability.list) {
-                        if (element.status === 'Failed') {
-                            isScanFailed = true
-                            break
-                        } else if (element.status === 'Progressing') {
-                            isScanInProgress = true
-                        }
-                    }
-                }
-            }
-            setLastExecutionDetail({
-                imageScanDeployInfoId: result.imageScanDeployInfoId,
-                severityCount: isSecurityScanV2Enabled ? getSecurityScanSeveritiesCount(result) : result.severityCount,
-                isError: false,
-                isScanned: isSecurityScanV2Enabled ? result.scanned : true,
-                isScanFailed,
-                isScanInProgress,
-            })
-        } catch (error) {
-            setLastExecutionDetail({
-                imageScanDeployInfoId: 0,
-                severityCount: { critical: 0, high: 0, medium: 0, low: 0, unknown: 0 },
-                isError: true,
-                isScanned: true,
-                isScanFailed: false,
-                isScanInProgress: false,
-            })
-        } finally {
-            setAPILoader(false)
-        }
-    }
-    useEffect(() => {
-        getScanData()
-    }, [])
+    const imageScanErrored = scanResultError || executionDetailsError
+    const imageScanFailed = scanResultResponse?.result.codeScan.status === 'Failed' ?? false
```

Comment:
Check if enums are present?

Reply:
Also check if we need executionDetailsResponse here

Reply:
not supported in v1 api